theano/gof/destroyhandler.py

```diff
@@ -945,6 +945,20 @@ def orderings(self, fgraph):
                 raise InconsistencyError("Attempting to destroy indestructible variables: %s" %
                         illegal_destroy)
 
+            # We want to compute a list of nodes that may never be computed,
+            # so that we do not force their computations (they may be uncomputeable).
+            # For that, we start by a list of lazy nodes (ifelse), and use io_toposort
+            # to find out which variables will always be computed, even if the lazy
+            # ones are not.
+            lazy_outputs = [var for var in fgraph.variables
+                            if var.owner and isinstance(var.owner, theano.ifelse.IfElse)]
+            always_computed = graph.io_toposort(fgraph.outputs, lazy_outputs)
```

Comment:
I don't understand why setting the fgraph.outputs as the inputs parameter is useful. I think fgraph.outputs should be fgraph.inputs. I also think that setting it as an empty list is equivalent and would be faster.

But I think this logic won't work if some of the ifelse depend on another one.